pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -524,14 +524,17 @@ func (p *PriorityQueue) MoveAllToActiveQueue() {
 		if p.isPodBackingOff(pod) {
 			if err := p.podBackoffQ.Add(pInfo); err != nil {
 				klog.Errorf("Error adding pod %v to the backoff queue: %v", pod.Name, err)
+			} else {
```

Comment:
We need better handling of errors, but `p.unschedulableQ.clear()` has much better performance compared to deleting pods individually. Since probability of errors when adding pods to other queues is low, we should optimize for the "no error" case.
One idea is to keep pods that an error has occurred for in a temporary slice. After the loop we still call  `p.unschedulableQ.clear()` and then iterate over the slice to add those pods back to the unschedulable queue.

Reply:
get it, done, thanks